### PR TITLE
[READY] Agent Stealth Box / Stealth Manual

### PR DESCRIPTION
**Pull request by Militaires**
:cl: Basilman
add: Added the stealth manual to the uplink, costs 8 TC. Find it under the implant section
/:cl:

This is a traitor item that allows you to deploy inside box that promptly turns invisible. If the box collides with a living being or vice versa, it will be temporarily revealed, pushing against the box will continue revealing it.

While the box is revealed, clicking on it will open the box as well as destroy it, clicking the action button while the box is deployed also destroys the box. **Finally, you can't open airlocks while inside the box**, that means you must exit the box > open > wait until cooldown > enter box if you're not a smart agent and wait for someone to open the door before sneaking behind them.

The box can be recognized as of syndicate nature due to a red label on its side.

also featuring really nice action button

### balance flock silencer: box vs chameleon

**disadvantages:**

- the box is actually less stealthy than the chameleon projector when it is in the middle of a crowd
- the box is MUCH more expensive than the chameleon (2 TC vs EIGHT TC, a huge difference)
- the box does not let you do anything while you are inside, **as in you cant shoot guns or open airlocks - without leaving the box** **(if you play at all, you will realize that this is the box's biggest flaw)**

**advantages:**

- doesnt need an item to copy like the chameleon
- generally more stealthy because full invisibility unlike the chameleon
- can let you go undetected even if you're found out once or twice, unlike the chameleon
- box is an innate ability which can not be taken or stolen from you, unlike the chameleon
- the box moves faster than the chameleon projector
- you're virtually invisible if you make sure to avoid people bumping into you, and not even borgs or the AI can detect you.

looks pretty balanced to me

### CODE REVIEW

**Comment by Kawaii-Big-Boss** [comment]
Name one bad PR that Militaries has made. I'll wait.
also lower the cost that's a bit much

**Comment by Militaires** [comment]
@Kawaii-Big-Boss that's how you gotta price these things to get em merged...............................................................

**Comment by GhostEcho** [comment]
holy fuck this is beautiful

**Comment by Militaires** [comment]
MAINTAINERS WILL LET YOU ME PRICE IT AT 8 TC ASKING FOR A FRIEND

**Comment by Kawaii-Big-Boss** [comment]
Having a high price so catgirl maintainers will merge it would explain the worthless 5 TC beartrap at least.
8 TC sounds incredible though. Getting an ebow and a box and sneaking around stunning everyone like true Metal Gear.

**Comment by subject217** [comment]
>get an ebow and a box

hey uh basilman can we make that 9 tc

**Comment by Militaires** [comment]
ok ill make it 9

File: code/game/objects/items/stealthmanual.dm [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+/datum/action/innate/agent_box/Activate()
+	if(!box)
+		if(cooldown < world.time - 30)
+			box = new(get_turf(owner))
+			owner.forceMove(box)
+			cooldown = world.time
+	else
+		owner.forceMove(get_turf(box))
+		qdel(box)
+		box = null
```
Comment (cyclowns) [inline code/game/objects/items/stealthmanual.dm]: QDEL_NULL(box)

**Comment by kevinz000** [comment]
:-1: I'm very against anything that grants full invis, even with the slowdown.

**Comment by subject217** [comment]
the point was that full invis + ebow is silly and it should specifically be 9 so that you could not do this strat

**Comment by Bluespace** [comment]
Why get this when you can get cham projector instead? Why would anyone pay 8tc for this when a click destroys it?

**Comment by Supermichael777** [comment]
At least make it able to convert lesser boxes with some time

**Comment by Nabski89** [comment]
Please look into the bug where trying to move diagonally at a wall in a box allows you to move at normal walking speed.

This is really strong without that fixed, and would be really hard to tell when someone is abusing that bug.

**Comment by ExcessiveUseOfCobblestone** [comment]
Add an overlay for ghosts-only, maybe a really big ladder icon :^)?

**Comment by ExcessiveUseOfCobblestone** [comment]
@Militaires the reason he wants it 9 is so that you explicitly _can't_ buy ebow with it.

**Comment by YakumoChen** [comment]
not to rain on the parade but isn't this just a worse chameleon projector?

**Comment by Tlaltecuhtli** [comment]
add cqc kills from being next to the box or add more bees

**Comment by Militaires** [comment]
this doesnt require an item to copy, goes invisible back again after being caught, and can be made on the spot, which makes it kind of different from the cham projector.

you cant ebow people from inside the box and you have to leave it + wait for the cooldown to go back, also by your own argument a cham projector + ebow would work just as well if not better and for cheaper TC so you really cant tell me it's OP because of that.

@Bluespace it's an ability, if the box is destroyed you can just make it again after a short cooldown

**Comment by Militaires** [comment]
it appears people dont realize this is an ability not an actual box, if the box is destroyed you can just click the ability again to get it back

**Comment by Militaires** [comment]
also kevinz is a buttbaby and im gonna suplex him

**Comment by Militaires** [comment]
### due to popular demand for a differentiating advantage, the box now moves at speed lower than run speed but slightly faster than walk speed

**Comment by ExcessiveUseOfCobblestone** [comment]
I thought the advantage was that it's an ability?

Ironically from the comments of the PR, this is a better chameleon projector BEFORE the buff.

**Comment by Militaires** [comment]
@ExcessiveUseOfCobblestone i did it cause "reeeeeeeeeeeeeeeeeeeeeeeeeeee cham proj is better"

**Comment by ExcessiveUseOfCobblestone** [comment]
lol you can't go "> people think this is a one use item vs an ability" then turn around and use their responses as justification for buffing it as if it was a one use item. Bad bad!

**Comment by zamolxius** [comment]
do you get tased if you get shot while in the box?  this could work as a panic projectile shield button. 

do AIs see you?  as in npcs,  space carp etc

**Comment by Militaires** [comment]
the box behaves exactly like a normal closed box aside from the invisibility, simple AIs can see you but i dont know if they'll do anything to a normal box out in the open

**Comment by ExcessiveUseOfCobblestone** [comment]
@Bluespace @YakumoChen did you realize this was a spell vs an item and, more importantly, do you think it should be buffed knowing it's a spell?

**Comment by Militaires** [comment]
before posting refer to the balance flock silencer posted exclusively for cobby

**Comment by ExcessiveUseOfCobblestone** [comment]
Your post after the fact has literally 0 bearing on how they took the feature initially.

If you can't even read my very simple question properly how do you expect people who don't code to read code, or to pick up visual cues from 2 gifs about the feature?

**Comment by Militaires** [comment]
**"and, more importantly, do you think it should be buffed knowing it's a spell?"**, cobby come back when you have an actual argument for this being overpowered instead of "he said she said"

**Comment by Militaires** [comment]
@MrDoomBringer injectors are gay, the implication here is that the manual is just telling you to put a box on and everyone believes it cause they're dumb. trying to explain everything with technology ruins it

**Comment by MrDoomBringer** [comment]
"everyone believes it cause they're dumb" uuh but the ai and 
ok

**Comment by Militaires** [comment]
@MrDoomBringer the AI and everyone actually see the box but they say "yeap that's a box" and ignore it so it's invisible on your screen

### D E E P  L O R E

**Comment by Bluespace** [comment]
"@Bluespace @YakumoChen did you realize this was a spell vs an item and, more importantly, do you think it should be buffed knowing it's a spell?"

I originally thought it was an item.
Knowing it's a spell, the OP looks decently balanced, but the meat and bones of balance will revolve around the box being used to ambush and ebow people. 

It does seem odd to have it as a spell though. Am I just conjuring boxes out of mid-air when i'm naked in perma?
A bluespace implant that can project a holographic box around you seems better, especially if clicking it vanishes it from existance.

**Comment by Militaires** [comment]
>It does seem odd to have it as a spell though. Am I just conjuring boxes out of mid-air when i'm naked in perma?
>A bluespace implant that can project a holographic box around you seems better, especially if clicking it vanishes it from existance.

i get what you mean by it not making much sense but honestly i dont think it matters really, whether it's a manual or **b**lue**s**pace [:^)] is just flushh tbhqfam brother

**Comment by optimumtact** [comment]
make it an implant you dumb head

**Comment by Militaires** [comment]
@ExcessiveUseOfCobblestone get over yourself
@optimumtact god fucking damn it ok

**Comment by Militaires** [comment]
now the stealth manual is a stealth implant per @optimumtact 's demand

**Comment by Militaires** [comment]
travis is being autistic but i dont know what's wrong, i was told that travis has been acting up lately and shitting itself for no reason though

**Comment by sawrge** [comment]
it owns

**Comment by Militaires** [comment]
it has the sawrge approval you guys gotta merge it now....

**Comment by McBawbaggings** [comment]
@Militaires permission for me to port this to hippie if it gets denied here?

File: code/game/objects/items/implants/implant_stealth.dm [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+/obj/structure/closet/cardboard/agent
+	name = "inconspicious box"
+	desc = "It's so normal that you didn't notice it before."
+	icon_state = "agentbox"
+	move_speed_multiplier = 0.5
+
+/obj/structure/closet/cardboard/agent/Initialize()
+	. = ..()
+	START_PROCESSING(SSfastprocess, src)
```
Comment (optimumtact) [inline code/game/objects/items/implants/implant_stealth.dm]: why are you using a subsystem ticker? animate the alpha when it updates with a timer if necessary, you're ticking for no reason most of the time.

**Comment by Enzarel** [comment]
Now i can finally fully live out my solid snake ops fantasies. Thank you mr. coder man

**Comment by Militaires** [comment]
@McBawbaggings of course

**Comment by Militaires** [comment]
@optimumtact done

**Comment by Militaires** [comment]
the box will always cause people to be alerted when it is opened, unlike a normal cardboard box

**Comment by Kagus** [comment]
So I just checked and both wiki and git are actually in agreement that the chameleon projector is 7 TC, not 2 TC. I also thought it was 2 TC. I can't find any recent PR or history showing someone changing it from 2 to 7.

Are we all just incredibly fucking blind, or was the TC cost of projectors actually changed recently?

**Comment by MrDoomBringer** [comment]
image
@Kagus >implying literally anyone on this repo actually plays the game

**Comment by 00aquir** [comment]
invisibly ambushing with weapons and moving is still op, people can just get rid of whole groups with that.

**Pull request merged**